### PR TITLE
[LoongArch64-RISCV64] Refactor the profiler for LoongArch64 and also fix some errors for RISCV64.

### PR DESCRIPTION
**Pull request by shushanhf**
This PR is part of the issue https://github.com/dotnet/runtime/issues/69705 to amend the LA's port.

Refactor the profiler for LoongArch64 and also fix some errors for RISCV64  after #91595

### CODE REVIEW

**Review by tomeksowi — APPROVED** [review]
RISC-V changes look good to me, thanks for the stack alignment fix.

File: src/coreclr/vm/riscv64/asmconstants.h [hunk 1]
```diff
@@ -255,7 +255,10 @@ ASMCONSTANTS_C_ASSERT(CallCountingStubData__TargetForThresholdReached == offseto
 #define PROFILE_LEAVE    2
 #define PROFILE_TAILCALL 4
 
-#define SIZEOF__PROFILE_PLATFORM_SPECIFIC_DATA 312
+// NOTE: this should be 16-byte aligned as stack size.
+#define SIZEOF__PROFILE_PLATFORM_SPECIFIC_DATA  0x140
```
Comment (tomeksowi) [inline src/coreclr/vm/riscv64/asmconstants.h:259]: Suggestion:
```C
ASMCONSTANTS_C_ASSERT(SIZEOF__PROFILE_PLATFORM_SPECIFIC_DATA % 16 == 0)
```

Comment (shushanhf) [reply src/coreclr/vm/riscv64/asmconstants.h:259]: Thanks very much.
Very good.
I will add it.

**Review by clamp03 — APPROVED** [review]
Thank you.

File: src/coreclr/jit/codegenloongarch64.cpp [hunk 2]
```diff
@@ -8677,10 +8649,78 @@ void CodeGen::genProfilingEnterCallback(regNumber initReg, bool* pInitRegZeroed)
...
+    assert((helper == CORINFO_HELP_PROF_FCN_LEAVE) || (helper == CORINFO_HELP_PROF_FCN_TAILCALL));
+
+    if (!compiler->compIsProfilerHookNeeded())
+    {
+        return;
+    }
+
+    compiler->info.compProfilerCallback = true;
+
+    if (compiler->compProfilerMethHndIndirected)
```
Comment (shushanhf) [inline src/coreclr/jit/codegenloongarch64.cpp:8702]: @jakobbotsch 
Can you give me some help?
`if (compiler->compProfilerMethHndIndirected)` , is this used under AOT's relocation ? which will be recomputed before running liking AOT ?

**Comment by shushanhf** [comment]
If don‘t add the `update the args registers within the GenerateProfileHelper`， the origin patch of this PR is OK while you run the `RunTest` within the `src/tests/profiler/unittest/inlining.cs`.

The `update the args registers within the GenerateProfileHelper` is used to fix the error about the `args` within the `Main(string[] args)`.

Now the related profiler tests is OK.

@jkotas 
Could you please review this PR?
Thanks very much!

File: src/coreclr/jit/codegenloongarch64.cpp (hunk 2 above)
Comment (jakobbotsch) [reply src/coreclr/jit/codegenloongarch64.cpp:8702]: It looks like dead code nowadays, I think you can just assert that it is false.

Comment (shushanhf) [reply src/coreclr/jit/codegenloongarch64.cpp:8702]: Thanks very for your help!

**Review by jkotas — APPROVED** [review]
LGTM

**Review by davmason — APPROVED** [review]

**Pull request merged**